iothub/device/src/ClientOptions.cs

```diff
@@ -54,5 +54,11 @@ public class ClientOptions
         /// or the <see cref="ModuleClient.CreateFromEnvironmentAsync(ClientOptions)"/> flow.
         /// </remarks>
         public int SasTokenRenewalBuffer { get; set; }
+
+        /// <summary>
+        /// A keep-alive for when using web sockets.
+        /// </summary>
+        /// <seealso href="https://docs.microsoft.com/dotnet/api/system.net.websockets.clientwebsocketoptions.keepaliveinterval"/>
+        public TimeSpan WebSocketKeepAlive { get; set; }
```

Comment:
James suggested adding it to `ClientOptions` as it applies to MQTT and AMQP.

Reply:
I like consolidating similar settings in a single holder like ClientOptions, but historically the .NET SDK has had transport level settings configured in the corresponding transport settings file. Maybe we could add it to our ITransportSettings interface to avoid duplicating code and still preserve the existing flow of setting all transport level settings in a single place?

Reply:
We should also call out explicitly that this keep-alive is different from mqtt keep-alive or amqp idle timeout.

Reply:
What would that mean for HTTP1 transport, I wonder?

Reply:
That's true, yes. Didn't we recently add a keep-alive for http clients so that we could trigger faster DNS refreshes? Do you remember if we made it configurable and if yes, where that property is set by users?

Reply:
I would still vote for us to duplicate the property and mqtt and amqp transport settings rather than putting it in client options. Given the way the sdk is set up today I don't think it would be intuitive for users to look at client options for configuring transport level settings.

Reply:
While I do like them to be consolidated, I have to agree that if we've done it a specific way before we shouldn't break that convention.